### PR TITLE
Monorepo filtering docs update

### PR DESCRIPTION
**Pull request by alisdair**
Changes:

- Update workspaces API docs
- Move the beta monorepo filtering doc into the version control settings page
- Update workspace create docs and screenshots
- Update references to monorepo-related issues to point to the new settings

### CODE REVIEW

File: content/source/docs/enterprise/workspaces/repo-structure.html.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ The upside of this approach is that it requires fewer files and runs fewer plans
 
 For organizations that have significant differences between environments, or prefer short-lived branches that are frequently merged into the master branch, we recommend creating a separate directory for each environment.
 
-~> **Important:** Since workspaces will queue a plan whenever their VCS branch changes or receives a pull request, you should avoid connecting too many workspaces to the same repository. Connecting more than about ten workspaces to one branch of a repo can cause slow performance and unnecessary noise; if you need to support more workspaces than that, we recommend splitting components into multiple repos.
+-> **Note:** By default, TFE will only trigger runs when the contents of the workspace's working directory changes. You can configure your workspace to also trigger runs when modules or other dependencies change. See [Automatic Run Triggering](../workspaces/settings.html#automatic-run-triggering) settings for more details.
```
[the remark this answers is not in the document]
Comment (alisdair) [reply content/source/docs/enterprise/workspaces/repo-structure.html.md]: I'm not sure what you mean. Modules/other dependencies was intended to mean "the other directories outside your working directory that should trigger runs". How did it read to you?

File: content/source/docs/enterprise/api/workspaces.html.md [hunk 2]
```diff
@@ -58,15 +58,17 @@ Key path                                      | Type    | Default   | Descriptio
...
 `data.attributes.migration-environment`       | string  | (nothing) | The legacy TFE environment to use as the source of the migration, in the form `organization/environment`. Omit this unless you are migrating a legacy environment.
-`data.attributes.queue-all-runs`              | boolean | false     | Whether runs should be queued immediately after workspace creation. When set to false, runs triggered by a VCS change will not be queued until at least one run is manually queued.
+`data.attributes.queue-all-runs`              | boolean | `false`   | Whether runs should be queued immediately after workspace creation. When set to false, runs triggered by a VCS change will not be queued until at least one run is manually queued.
 `data.attributes.terraform-version`           | string  | (nothing) | The version of Terraform to use for this workspace. Upon creating a workspace, the latest version is selected unless otherwise specified (e.g. `"0.11.1"`).
+`data.attributes.trigger-prefixes`            | array   | `[]`      | List of repository-root-relative paths which list all locations to be tracked for changes.
```
[the remark this answers is not in the document]
Comment (alisdair) [reply content/source/docs/enterprise/api/workspaces.html.md]: Updating to:

> List of repository-root-relative paths which should be tracked for changes, in addition to the working directory.

File: content/source/docs/enterprise/api/workspaces.html.md [hunk 3]
```diff
@@ -105,6 +107,30 @@ _With a VCS repository_
 }
 ```
 
+_For a monorepo_
+
+This workspace will be triggered by changes in any of the specified directories: `/networking`, `/modules`, or `/vendor`.
```
[the remark this answers is not in the document]
Comment (alisdair) [reply content/source/docs/enterprise/api/workspaces.html.md]: Rewording to:

> A run will be triggered in this workspace when changes are detected in any of the specified directories: `/networking`, `/modules`, or `/vendor`.

File: content/source/docs/enterprise/workspaces/settings.html.md [hunk 4]
```diff
@@ -114,6 +114,25 @@ You can use the "Select a VCS connection" buttons and "Repository" field to chan
...
+
+You can adjust this behavior in two ways:
+
+- **Add more trigger directories.** TFE will queue runs for changes in any of the specified trigger directories (including the working directory).
+
+    For example, if you use a top-level `modules` directory to share Terraform code across multiple configurations, changes to the shared modules are relevant to every workspace that uses that repo. You can add `modules` as a trigger directory for each workspace to make sure they notice any changes to shared code.
+- **Mark the entire repository as relevant.** If you set the "Automatic Run Triggering" setting to "Always Trigger Runs," TFE will assume that anything in the repository might affect the workspace's configuration, and will queue runs for any change.
+
+    This can be useful for repos that don't have multiple configurations but require a working directory for some other reason. It's usually not what you want for true monorepos, since it queues unnecessary runs and slows down your ability to provision infrastructure.
```
Comment (beekus) [inline content/source/docs/enterprise/workspaces/settings.html.md]: Do people reading this know what a `monorepo` is? Should we link to the [repo structure](https://www.terraform.io/docs/enterprise/workspaces/repo-structure.html#directories) section (and possibly link back here?

Comment (alisdair) [reply content/source/docs/enterprise/workspaces/settings.html.md]: I think monorepo is probably well-understood, at least with the extra context, and I can't find a good way to link to the repo structure page from here. I did add a link from the repo structure page back to this one, though.

**Comment by beekus** [comment]
The content in `### Automatic Run Triggering` is nice and clear.

**Comment by alisdair** [comment]
> The content in `### Automatic Run Triggering` is nice and clear.

That's because @nfagerlund wrote it 😅 🙌

**Pull request merged**